admin/broadleaf-open-admin-platform/src/main/resources/open_admin_style/js/admin/components/filterbuilder.js

```diff
@@ -907,6 +907,62 @@
                     $filterButton.closest('.main-content').find('.sticky-container .filter-text').hide();
                 }
             }
+
+            BLCAdmin.filterBuilders.updateAppliedFiltersContainerHtml(filterBuilder);
+        },
+
+        updateAppliedFiltersContainerHtml(filterBuilder) {
+            var $container = $('#applied-filters-container-' + filterBuilder.hiddenId);
+            if ($container.length) {
+                $container.html(this.buildAppliedFiltersContainerHtml(filterBuilder));
+            }
+        },
+
+        buildAppliedFiltersContainerHtml(filterBuilder) {
+            var ruleJson = JSON.parse($("#" + filterBuilder.hiddenId).val());
+            if (!ruleJson || !ruleJson.data || ruleJson.data.length === 0) {
+                return ""
+            }
+            var rules = ruleJson.data[0].rules;
+            if (!rules || rules.length === 0) {
+                return "";
+            }
+            if (rules.length === 1) {
+                return "<span class='single-filter'>" + asInnerHtml(filterBuilder, rules[0]) + "</span>";
+            }
+            var $list = $("<ul></ul>");
+            for (var i = 0; i < rules.length; i++) {
+                var item = "<li>" + asInnerHtml(filterBuilder, rules[i]) + "</li>";
+                $list.append(item);
+            }
+            return $list.prop('outerHTML');
+            
+            function asInnerHtml(filterBuilder, rule) {
+                var label = "<b>" + filterBuilder.getFieldLabelById(rule.id) + "</b>";
+                var operator = filterBuilder.getOperatorLabelByOperatorType(rule.operator);
+                var value = "";
+
+                switch(rule.operator) {
+                    case "IS_NULL":
+                        break;
+                    case "BETWEEN":
+                        var arr = JSON.parse(rule.value);
+                        value = "<b>" + arr[0] + " AND " + arr[1] + "</b>";
+                        break;
+                    case "COLLECTION_IN":
+                    case "COLLECTION_NOT_IN":
+                        var arr = JSON.parse(rule.value);
```

Comment:
@hasheniuk, to get the name for the collection-in/not-in, 
1. Grab the entity context path from `$('#listGrid-main-header).data('path')`
2. Grab the field name from the rule (`rule.field`)
3. Create a new endpoint in `AdminBasicEntityController` like `/selectize/name`
4. Send the field name and value (e.g., defaultCategory:2001) to the new endpoint
5. Add a method in `AdminEntityService` that can take the sectionKey (/admin/product), the field name (defaultCategory), and the ID to return the object
6. then take the name from it 
7. and return it

I think you may need to add `blDynamicEntityDao` to the AdminEntityService and add a method to it as well that works like `DynamicEntityDao#readOtherEntitiesWithPropertyValue()` but returns the associated entity/value for the field in question instead of a list of IDs

Reply:
@jfleschler Does that sound about right?

Reply:
We will only use this method for `COLLECTION_IN` or `COLLECTION_NOT_IN`, so when we build the criteria query we can do the following:

1. Get the classname for the sectionKey
2. Get the Impl class for the classname (see `DynamicEntityDao#getImplClass()`
3. Use reflection to get the Impl class (ProductImpl) of the field (defaultCategory -> CategoryImpl)
4. Setup a CriteriaQuery = builder.createQuery(CategoryImpl)
5. Root = criteria.from(ProductImpl)
6. Restriction = builder.in(root.get(fieldName).get("id"), <list of ids from the filter>)
etc.